app/controllers/admin/application_controller.rb

```diff
@@ -0,0 +1,19 @@
+# All Administrate controllers inherit from this
+# `Administrate::ApplicationController`, making it the ideal place to put
+# authentication logic or other before_actions.
+#
+# If you want to add pagination or other controller-level concerns,
+# you're free to overwrite the RESTful controller actions.
+module Admin
+  class ApplicationController < Administrate::ApplicationController
+    include ActionController::HttpAuthentication::Basic::ControllerMethods
+
+    before_action :authenticate_admin
+
+    def authenticate_admin
+      authenticate_or_request_with_http_basic do |given_name, given_password|
+        ApiConsumer.exists?(public_api_key: given_name, private_api_key: given_password)
```

Comment:
Initially the idea was that `public_api_key` is a consumers public name for read access and `private_api_key` is for writing only. Not sure if this concept still holds.

Reply:
Yep, this is still being used that way. Right now we have 1 external user who uses a public-only key. Additionally, the web application and the apps must contain a hardcoded key, that one is read-only as well.

I'd like to keep the differentiation between readonly and read/write keys. The admin interface should only be accessible using the read/write (private) API key.

Would it be hard to slap a session based login page in front of the admin interface? Basic auth has its downsides, for example: You cannot clear a basic auth session without closing and re-opening the browser. It's not that important though, we can keep the basic auth login for now. What about using the contact_email field for the username? Or just the string "admin"?

Reply:
Sure, I can do that.